### PR TITLE
Feature: Profit Per Excavation

### PR DESCRIPTION
**Pull request by hannibal002**
## What
Add Profit Per Excavation.

<details>
<summary>Images</summary>
image

</details>

## Changelog New Features
+ Added Profit Per Excavation. - hannibal2
    * Show profit/loss in chat after each excavation.
    * Also include breakdown information on hover.

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ProfitPerExcavation.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+class ProfitPerExcavation {
+    private val config get() = SkyHanniMod.feature.mining.fossilExcavator
+
+    @SubscribeEvent
+    fun onFossilExcavation(event: FossilExcavationEvent) {
+        if (!config.profitPerExcavation) return
+        val loot = event.loot
+
+        val scrapItem = "SUSPICIOUS_SCRAP".asInternalName()
```
Comment (j10a1n15) [inline src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ProfitPerExcavation.kt]: Probably move this line before the event, in like 17 or smth

**Comment by ItsEmpa** [comment]
shouldnt changelog say "Added" not adding

**Comment by ItsEmpa** [comment]
also whats up with it saying -12 coins for fossil dust

image

File: src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ProfitPerExcavation.kt (hunk 1 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ProfitPerExcavation.kt]: fixed

**Comment by hannibal002** [comment]
> also whats up with it saying -12 coins for fossil dust
> 
> image

hypixel added a placeholder item, wow :O
image

File: src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ProfitPerExcavation.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                val profit = amount * it.getPrice()
+                val text = "Found $name §8${amount.addSeparators()}x §7(§6${NumberUtil.format(profit)}§7)"
+                map[text] = profit
+                totalProfit += profit
+            }
+        }
+
+        val scrapItem = FossilExcavatorAPI.scrapItem
+
+        val scrapPrice = scrapItem.getPrice()
```
Comment (j10a1n15) [inline src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ProfitPerExcavation.kt:37]:
```suggestion
        val scrapPrice = FossilExcavatorAPI.scrapItem.getPrice()
```

imo a bit cleaner

**Review by j10a1n15 — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ProfitPerExcavation.kt (hunk 2 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ProfitPerExcavation.kt:37]: We use it twice
image

not sure if this is truly cleaner

Comment (j10a1n15) [reply src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ProfitPerExcavation.kt:37]: Oh didn't see this
nvm then

**Review by CalMWolfs — APPROVED** [review]
code looks fine, cant test

**Pull request merged**